c10/util/Half.h

```diff
@@ -429,28 +429,34 @@ C10_CLANG_DIAGNOSTIC_IGNORE("-Wimplicit-float-conversion")
 // `error: comparison of constant '255' with boolean expression is always false`
 // for `f > limit::max()` below
 template <typename To, typename From>
-std::enable_if_t<std::is_same_v<From, bool>, bool> overflows(From /*f*/) {
+std::enable_if_t<std::is_same_v<From, bool>, bool> overflows(
+    From /*f*/,
+    bool strict_unsigned = false) {
   return false;
 }
 
 // skip isnan and isinf check for integral types
 template <typename To, typename From>
 std::enable_if_t<std::is_integral_v<From> && !std::is_same_v<From, bool>, bool>
-overflows(From f) {
+overflows(From f, bool strict_unsigned = false) {
   using limit = std::numeric_limits<typename scalar_value_type<To>::type>;
-  if (!limit::is_signed && std::numeric_limits<From>::is_signed) {
+  if (!strict_unsigned && !limit::is_signed &&
+      std::numeric_limits<From>::is_signed) {
     // allow for negative numbers to wrap using two's complement arithmetic.
     // For example, with uint8, this allows for `a - b` to be treated as
     // `a + 255 * b`.
     return greater_than_max<To>(f) ||
-        (c10::is_negative(f) && -static_cast<uint64_t>(f) > limit::max());
+        (c10::is_negative(f) &&
+         -static_cast<uint64_t>(f) > static_cast<uint64_t>(limit::max()));
```

Comment:
This resulted in massive warnings as nvcc can't figure out that if can be short-circuited if TO is a floating point type.
Following fixes it: https://github.com/pytorch/pytorch/pull/117023

Reply:
Thank you haha